mage_integrations/mage_integrations/tests/sources/test_base.py

```diff
@@ -0,0 +1,473 @@
+                ),
+                type=dict(
+                    type=['null', 'string'],
+                ),
+                isdeleted=dict(
+                    type=['null', 'boolean'],
+                ),
+                actionid=dict(
+                    type=['null', 'string'],
+                ),
+                createdbyid=dict(
+                    type=['null', 'string'],
+                ),
+                systemmodstamp=dict(
+                    type=['null', 'string'],
+                ),
+                lastmodifieddate=dict(
+                    type=['null', 'string'],
+                ),
+                createddate=dict(
+                    type=['null', 'string'],
+                    format='date-time',
+                ),
+                id=dict(
+                    type=['null', 'string'],
+                ),
+                name=dict(
+                    type=['null', 'string'],
+                ),
+                airecordinsightid=dict(
+                    type=['null', 'string'],
+                ),
+                lastmodifiedbyid=dict(
+                    type=['null', 'string'],
+                ),
+            ),
+            type='object',
+        ),
+        stream='demo_table',
+        tap_stream_id='demo_table',
+        unique_conflict_method='UPDATE',
+        unique_constraints=[],
+    )
+    stream2 = dict(
+        auto_add_new_fields=True,
+        bookmark_properties=['id'],
+        key_properties=[],
+        metadata=demo_users_stream_metadata,
+        replication_method='INCREMENTAL',
+        schema=dict(
+            properties=dict(
+                age=dict(type=['null', 'integer']),
+                id=dict(type=['null', 'string']),
+                last_name=dict(type=['null', 'string']),
+                first_name=dict(type=['null', 'string']),
+                color=dict(type=['null', 'string'])
+            ),
+            type='object',
+        ),
+        stream='demo_users',
+        tap_stream_id='demo_users',
+        unique_conflict_method='UPDATE',
+    )
+
+    return [stream1, stream2]
+
+def build_sample_streams_catalog():
+    return Catalog.from_dict(
+        dict(streams=build_sample_streams_list())
+    )
+
+class BaseSourceTests(unittest.TestCase):
+    maxDiff = None
+
+    def test_templates(self):
+        # Template folders exist in the different integration source folders.
+        source = PostgreSQL()
+        templates = source.templates()
+        self.assertEqual(
+            templates,
+            {
+                'config': {
+                    'database': '',
+                    'host': '',
+                    'password': '',
+                    'port': 5432,
+                    'schema': '',
+                    'username': '',
+                    'replication_slot': '',
+                }
+            },
+        )
+
+    def test_discover(self):
+        # Testing with Intercom source, since it has a "schemas"
```

Comment:
For multiline comments, the recommended style is 
```
"""
Testing with Intercom source, since it has a "schemas"
folder and no "discover" subclass method.
"""
```